### PR TITLE
Relax IScheduledExecutor stats_manyRepetitionsTask assertions

### PR DESCRIPTION
**Pull request by tkountis**
Fixes https://github.com/hazelcast/hazelcast/issues/12103

Minor JavaDoc cleanup.

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/scheduledexecutor/ScheduledExecutorServiceSlowTest.java [hunk 1]
```diff
@@ -139,12 +139,10 @@ public void stats_manyRepetitionsTask() throws Exception {
                 new ICountdownLatchRunnableTask("latch"), 0, 10, SECONDS);
 
         latch.await(120, SECONDS);
-        // wait for run-cycle to finish before cancelling, in order for stats to get updated
-        sleepSeconds(4);
         future.cancel(false);
 
         ScheduledTaskStatistics stats = future.getStats();
-        assertEquals(6, stats.getTotalRuns());
+        assertEquals(6, stats.getTotalRuns(), 1);
```
[the remark this answers is not in the document]
Comment (tkountis) [reply hazelcast/src/test/java/com/hazelcast/scheduledexecutor/ScheduledExecutorServiceSlowTest.java]: yes

File: hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/ScheduledExecutorContainer.java [hunk 2]
```diff
@@ -340,8 +340,7 @@ void checkNotAtCapacity() {
     }
 
     /**
-     * State is published after every run. When replicas get promoted, they start of, with the latest state see {@link
-     * TaskRunner#initOnce()}
+     * State is published after every run. When replicas get promoted, they start of, with the latest state
```
Comment (Donnerbart) [inline hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/ScheduledExecutorContainer.java]: This sounds a bit more fluent to me:
> When replicas get promoted, they start with the latest state.

File: hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/ScheduledTaskDescriptor.java [hunk 3]
```diff
@@ -30,8 +30,13 @@
 import java.util.concurrent.atomic.AtomicReference;
 
 /**
- * Metadata holder for scheduled tasks. Scheduled ones have a reference in their future in {@link #future}.
- * Reserved ones have this reference null.
+ * Metadata holder for scheduled tasks.
+ * Active tasks, eg. not suspended, hold a non-null {@link #future} reference. Suspended ones, i.e., backups
+ * or on-going migration have {@link #future} set to null.
+ *
```
Comment (Donnerbart) [inline hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/ScheduledTaskDescriptor.java]: Missing `<p>` for a new paragraph.

File: hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/ScheduledTaskDescriptor.java [hunk 4]
```diff
@@ -40,21 +45,11 @@
...
-     * through partition threads. Reads are MP for member owned tasks.
-     */
-    private AtomicReference<ScheduledTaskResult> resultRef = new AtomicReference<ScheduledTaskResult>(null);
+    private volatile Map<?, ?> state;
 
-    /**
-     * SPMC
-     */
-    private Map<?, ?> state;
+    private final AtomicReference<ScheduledTaskResult> resultRef = new AtomicReference<ScheduledTaskResult>(null);
```
Comment (Donnerbart) [inline hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/ScheduledTaskDescriptor.java]: `private final` should be above a `private volatile`

**Comment by tkountis** [comment]
run-lab-run

**Pull request merged**